File: Examples/CSharp/ProgrammersGuide/Working-Shapes/SetXFormdata.cs
using System.IO;
using Aspose.Diagram;
namespace Aspose.Diagram.Examples.CSharp.Shapes
{
    public class SetXFormdata
    {
        public static void Run()
        {
            // ExStart:SetXFormdata
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Shapes();

            // call a Diagram class constructor to load the VSD diagram
            Diagram diagram = new Diagram(dataDir + "SetXFormdata.vsd");

            // Find a particular shape and update its XForm
            foreach (Aspose.Diagram.Shape shape in diagram.Pages[0].Shapes)
            {
                if (shape.NameU.ToLower() == "process" && shape.ID == 1)
                {
                    shape.XForm.PinX.Value = 5;
                    shape.XForm.PinY.Value = 5;
                }
            }
            // save diagram
            diagram.Save(dataDir + "SetXFormdata_Out.vsdx", SaveFileFormat.VSDX);
            // ExEnd:SetXFormdata
        }
    }
}

File: Examples/VisualBasic/Working-Shapes/SetXFormdata.vb
Imports Microsoft.VisualBasic
Imports System.IO
Imports Aspose.Diagram

Namespace Shapes
    Public Class SetXFormdata
        Public Shared Sub Run()
            ' ExStart:SetXFormdata
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Shapes()

            ' Call the diagram constructor to load diagram from a VDX file
            Dim vdxDiagram As New Diagram(dataDir & "SetXFormdata.vsd")

            ' Find a particular shape and update its XForm
            For Each shape As Aspose.Diagram.Shape In vdxDiagram.Pages(0).Shapes
                If shape.NameU.ToLower() = "process" AndAlso shape.ID = 1 Then
                    shape.XForm.PinX.Value = 5
                    shape.XForm.PinY.Value = 5
                End If
            Next shape
            vdxDiagram.Save(dataDir & "SetXFormdata_out.vsdx", SaveFileFormat.VSDX)
            ' ExEnd:SetXFormdata
        End Sub
    End Class
End Namespace